src/editor/input_center.cpp

```diff
@@ -432,7 +434,18 @@ EditorInputCenter::move_object() {
       dragged_object = NULL;
       return;
     }
-    dragged_object->move_to(sector_pos - obj_mouse_desync);
+    Vector new_pos = sector_pos - obj_mouse_desync;
+    if (snap_to_grid) {
+      auto& snap_grid_size = snap_grid_sizes[selected_snap_grid_size];
+      new_pos.x = int(new_pos.x / snap_grid_size) * snap_grid_size;
```

Comment:
I think you can use
new_pos = (new_pos / snap_grid_size) * snap_grid_size;

We have operators defined for vectors.